Update arenas #2

ikonomov opened the pull request:
- closed the arenas to allow phasing to function properly
- resized the arena maps to their minimum required area
- modified hell arena to match more closely the layout generated for level 16
- removed stairs up
- added /leavearena command to visit town from arenas (thank you KP!)

The maps are quite simple, primarily designed with functionality in mind.  Having only their entry point hard coded they can be very easily adopted as a template to edit and change according to player preference and then replace in their devilutionx/arena folder.

![cathedralarena](https://github.com/diasurgical/devilutionX/assets/5624187/7e6b1aad-0fc7-47bd-9ead-f8fce7ea6724)

![hellarena](https://github.com/diasurgical/devilutionX/assets/5624187/00c14771-5f68-4c2b-aec9-9b128841fadf)

![circleoflifearena](https://github.com/diasurgical/devilutionX/assets/5624187/8ab6d6e2-c887-4c85-a116-955aba4ccc8b)

ikonomov committed:
update arenas #2

Closed the areas to allow phasing to function properly.

kphoenix137 commented:
Perhaps the stairs and entrance areas should be removed entirely and you would use a command to exit the arena

ikonomov commented:
I think you are right.  I like it both ways, but without the stairs it would make it even more simplified which is even better I think and definitely more consistent with the command to get in.

ikonomov committed:
update

- added catacombs and caves arena
- enlarged all arena maps to 40x40 tiles
- removed stairs up
- renamed /arenapot command to /pots

Left to do:
- add /town command to visit town from arenas
- add /scrolls command to add scrolls of resurrect to inventory in arena

ikonomov committed:
add leavearena command

Thank you KP!

ikonomov committed:
Update catacombs.dun

ikonomov committed:
Update caves.dun

ikonomov committed:
Update catacombs.dun

missed some of the shadows on the floor tiles

ikonomov committed:
add shadow tiles #2

Added missing shadows from the hell and circle of life arenas.  Will blame this one on Constantine. XD

AJenbo commented on Source/control.cpp:
```diff
@@ -523,7 +525,7 @@ std::string TextCmdLevelSeed(const string_view parameter)
 	{ N_("/help"), N_("Prints help overview or help for a specific command."), N_("[command]"), &TextCmdHelp },
 	{ N_("/arena"), N_("Enter a PvP Arena."), N_("<arena-number>"), &TextCmdArena },
-	{ N_("/arenapot"), N_("Gives Arena Potions."), N_("<number>"), &TextCmdArenaPot },
+	{ N_("/pots"), N_("Gives Arena Potions."), N_("<number>"), &TextCmdArenaPot },
```
```suggestion
	{ N_("/arenapot"), N_("Gives Arena Potions."), N_("<number>"), &TextCmdArenaPot },
```

AJenbo commented:
Judging from the screenshots these levels do not have stairs and require a feature that might come in 1.6.0.

The file sizes are 5x the previous ones, maybe resize the leves to just the part that isn't dirt.

I would also prefer not adding new level types before 1.6.0, but that's not as important.

ikonomov committed:
Update Source/control.cpp

Co-authored-by: Anders Jenbo <anders@jenbo.dk>

ikonomov commented:
It's perfectly fine to delay this commit for 1.6.0 then.  The /leavearena command was requested feature that we agreed on having and there is no need to rush it.

I've expanded the size of the map to 40x40 tiles (the same size as town) on purpose in order to allow custom maps to be very easily placed later into the devilutionx/arena folder by gamers.  Having a larger map size allows much more flexibility for making custom maps.  I don't imagine that anybody would want anything bigger than that and having much smaller active area can be accommodated without any problems, as can be seen.  Without the stairs the only thing that is hardcoded outside those map files will be the entry point.  This way if somebody doesn't like the maps it would be very easy to change them without having to change any code.  That was the reason behind the 40x40 map size, having realistic expectations that my map designs and Constantine's will not and cannot be liked by everybody.  My hope is that in time we can upload somewhere a library of custom maps with screenshots where people can pick and download the ones that they prefer.

AJenbo commented:
The map size is dynamic, so it can be change independent of what the default map might have. So there aren't any actual advantages and it results in larger files.

ikonomov commented:
Hah, that is true!  I don't know why I got the idea that map size is hard coded, when it's just the entry and exit points that are.  I'll resize them back.  Also I keep looking at my caves and catacombs designs and I'm not quite happy with them, they don't offer anything unique.  I'm thinking about just pulling them out and making them part of a collection of custom arena maps, maybe on a Wiki page.  I'll be away for maybe a week or two, but will get to it whenever I have time.  Since we won't be merging this for 1.5.1 we can keep it open until we get it right.

ikonomov committed:
Merge branch 'diasurgical:master' into dev_arena

ikonomov committed:
update

- resized the arena maps to their minimum required area
- modified hell arena to match more closely the layout generated for level 16
- removed previously added catacombs and caves arenas

ikonomov commented:
Sorry, closed by mistake.  Everything should be done now, ready for review.

ikonomov commented:
I was hoping to create a #3 once this PR gets merged, but in case it doesn't here is a custom map that Shade shared which we can consider adding in the future.
![Custom_Dungeon_Layout](https://github.com/diasurgical/devilutionX/assets/5624187/b6a9a5ff-53dc-43a5-bfec-44d2efa1edd2)
[church.zip](https://github.com/diasurgical/devilutionX/files/12833137/church.zip)

AJenbo commented:
looks really cool. I don't think this PR is ready to merge in it's current state as the map relies on things that are not in DevilutionX

kphoenix137 commented:
Please add trigger text for when you hover over stairs. It's current missing. Also that map isn't very teleport and phasing friendly. That's an issue I was having with the revisions of the original arenas. I keep accidentally teleporting and phasing outside the arena area. Can this be addressed?

kphoenix137 commented:
Please add trigger text for when you hover over stairs. It's current missing. Also that map isn't very teleport and phasing friendly. That's an issue I was having with the revisions of the original arenas. I keep accidentally teleporting and phasing outside the arena area. Can this be addressed?

ikonomov commented:
Unless somebody objects I suggest we leave arenas after the Lua gets done.  Then we can have a "dueling" mod where we can really make something out of this idea.

kphoenix137 commented:
> Unless somebody objects I suggest we leave arenas after the Lua gets done. Then we can have a "dueling" mod where we can really make something out of this idea.

Disagree. We already have all the functionality for arenas in the code already, there is no sense in half-assing it, especially if Lua gets pushed back yet another update or two. I think we either should commit to quality with arenas or completely remove the functionality (and I'm not in favor of the latter option)

ikonomov commented:
I'm in favor of the latter option.

AJenbo commented:
We have Lua now thought...

kphoenix137 commented:
Now I'm in full support of moving it to Lua
